### PR TITLE
[CEL doc] Fix the hyperlinks, add quantity, optional, CrossTypeNumericComparisons which is available now

### PR DESCRIPTION
**Pull request by cici37**
Fix https://github.com/kubernetes/website/issues/44958

The quantity doc is cherry picked from https://github.com/kubernetes/website/pull/42454 authored by @alexzielenski 
The library is available in 1.29 together with couple other enabled func. 
I will be updating the other func available in 1.30 when dev-1.30 branch is ready. Thanks!

### CODE REVIEW

**Comment by cici37** [comment]
/cc @pohly @shubham-asati 
Thanks!

File: content/en/docs/reference/using-api/cel.md [hunk 1]
```diff
@@ -67,23 +67,17 @@ CEL is configured with the following options, libraries and language features, i
...
-| [Default UTC Time Zone](opt2)                       |                                                                                                                 | All Kubernetes versions    |
-| [Eagerly Validate Declarations](opt3)               |                                                                                                                 | All Kubernetes versions    |
-| [extended strings library](stringsgodoc), Version 1 | `charAt`, `indexOf`, `lastIndexOf`, `lowerAscii`, `upperAscii`, `replace`, `split`, `join`, `substring`, `trim` | All Kubernetes versions    |
+| [Standard macros](https://github.com/google/cel-spec/blob/v0.7.0/doc/langdef.md#macros)                        | `has`, `all`, `exists`, `exists_one`, `map`, `filter`                                                           | All Kubernetes versions    |
+| [Standard functions](https://github.com/google/cel-spec/blob/master/doc/langdef.md#list-of-standard-definitions)                        | See [official list of standard definitions](stdlib)                                                             | All Kubernetes versions    |
```
Comment (jpbetz) [inline content/en/docs/reference/using-api/cel.md:71]:
```suggestion
| [Standard functions](https://github.com/google/cel-spec/blob/master/doc/langdef.md#list-of-standard-definitions)                        | See [official list of standard definitions](https://github.com/google/cel-spec/blob/master/doc/langdef.md#list-of-standard-definitions)                                                             | All Kubernetes versions    |
```

File: content/en/docs/reference/using-api/cel.md [hunk 2]
```diff
@@ -67,23 +67,17 @@ CEL is configured with the following options, libraries and language features, i
...
 | Kubernetes authorizer library                       | See [Kubernetes authorizer library](#kubernetes-authorizer-library)                                             | All Kubernetes versions    |
 
-[stdmacros]: https://github.com/google/cel-spec/blob/v0.7.0/doc/langdef.md#macros
-[stdlib]: https://github.com/google/cel-spec/blob/master/doc/langdef.md#list-of-standard-definitions
-[stringsgodoc]: https://pkg.go.dev/github.com/google/cel-go/ext#Strings
-[opt1]: https://pkg.go.dev/github.com/google/cel-go@v0.17.4/cel#HomogeneousAggregateLiterals
-[opt2]: https://pkg.go.dev/github.com/google/cel-go@v0.17.4/cel#DefaultUTCTimeZone
-[opt3]: https://pkg.go.dev/github.com/google/cel-go@v0.17.4/cel#EagerlyValidateDeclarations
 [opt4]: https://pkg.go.dev/github.com/google/cel-go@v0.17.4/cel#OptionalTypes
 [opt5]: https://pkg.go.dev/github.com/google/cel-go@v0.17.4/cel#CrossTypeNumericComparisons
```
Comment (jpbetz) [inline content/en/docs/reference/using-api/cel.md:82]:
```suggestion
```

Comment (cici37) [reply content/en/docs/reference/using-api/cel.md]: I have updated those links into the table in the same PR :)

**Comment by pohly** [comment]
/lgtm

I checked the deploy preview. Now I know where I should have looked before :smile: Thanks!

**Comment by Shubham82** [comment]
Thanks @cici37 preview looks good to me.
https://deploy-preview-44972--kubernetes-io-main-staging.netlify.app/docs/reference/using-api/cel/
/lgtm

**Comment by reylejano** [comment]
Changes look good
Thank you Cici
/approve

**Pull request merged**